common/djangoapps/student/roles.py

```diff
@@ -134,7 +134,7 @@ def has_user(self, user):
         """
         Return whether the supplied django user has access to this role.
         """
-        if not (user.is_authenticated() and user.is_active):
+        if not (getattr(user, 'is_authenticated', True) and user.is_active):
```

Comment:
> it seems like this is supposed to be a method

Yep, so it is. I got off-track because I ended up looking at some 1.10 Django code that pointed to using it as a property instead (returning a fun CallableBool class so legacy code keeps working). I'll go ahead and do the shim to `lambda: True`, and patch the code in `roles.py` to call it as a method (it isn't doing so consistently right now, which didn't help my confusion about the situation).